include/deal.II/distributed/fully_distributed_tria.h

```diff
@@ -0,0 +1,406 @@
+     * distributes the coarse grid and gives flexibility regarding partitioning,
+     * the following ingredients are required:
+     * - a locally relevant coarse-grid triangulation
+     *   (vertices, cell definition; including a layer of ghost cells)
+     * - a mapping of the locally relevant coarse-grid triangulation into the
+     *   global coarse-grid triangulation
+     * - information about which cell should be refined as well as information
+     *   regarding the subdomain_id, the level_subdomain_id, manifold_id,
+     *   and boundary_id of each cell.
+     *
+     * The ingredients listed above are bundled in the struct
+     * parallel::fullydistributed::ConstructionData. The user has to fill this
+     * data structure - in a pre-processing step - before actually creating the
+     * triangulation. Predefined functions to create ConstructionData
+     * can be found in the namespace dealii::fullydistributed::Util.
+     *
+     * Once the ConstructionData `construction_data` has been constructed, the
+     * triangulation `tria` can be created by calling
+     * `tria.reinit(construction_data);`.
+     *
+     * @note This triangulation supports: 1D/2D/3D, hanging nodes,
+     *       geometric multigrid, and periodicity.
+     *
+     * @note Currently no modifications of the triangulation is supported after
+     *       it has been created, i.e., this type of mesh does not support
+     *       any form of adaptivity, not even simple global refinements and
+     *       coarsenings.
+     *
+     * @note Currently only simple periodicity conditions (i.e. without offsets
+     *       and rotation matrices - see also the documentation of
+     *       GridTools::collect_periodic_faces)) are supported.
+     *
+     * @author Peter Munch, 2019
+     */
+    template <int dim, int spacedim = dim>
+    class Triangulation
+      : public parallel::DistributedTriangulationBase<dim, spacedim>
+    {
+    public:
+      using cell_iterator =
+        typename dealii::Triangulation<dim, spacedim>::cell_iterator;
+
+      using active_cell_iterator =
+        typename dealii::Triangulation<dim, spacedim>::active_cell_iterator;
+
+      using CellStatus =
+        typename dealii::Triangulation<dim, spacedim>::CellStatus;
+
+      /**
+       * Configuration flags for fully distributed Triangulations to be set in
+       * the constructor. Settings can be combined using bitwise OR.
+       */
+      enum Settings
+      {
+        /**
+         * Default settings, other options are disabled.
+         */
+        default_setting = 0x0,
+        /**
+         * This flags needs to be set to use the geometric multigrid
+         * functionality. This option requires additional computation and
+         * communication.
+         */
+        construct_multigrid_hierarchy = 0x1
+      };
+
+      /**
+       * Constructor.
+       *
+       * @param mpi_communicator denotes the MPI communicator to be used for the
```

Comment:
```suggestion
       * @param mpi_communicator The MPI communicator to be used for the
```
(Take a look at how doxygen renders these `@param` statements.

Reply:
This was just copy-and-past from `p:d:t` (see also PR #8725).